sql/src/main/java/io/crate/planner/consumer/QueryAndFetchConsumer.java

```diff
@@ -99,25 +102,113 @@ public Plan visitQueriedTable(QueriedTable table, ConsumerContext context) {
         @Override
         public Plan visitQueriedSelectRelation(QueriedSelectRelation relation, ConsumerContext context) {
             QuerySpec qs = relation.querySpec();
-            if (qs.hasAggregates() || qs.groupBy().isPresent()) {
+            if (!isSimpleSelect(qs, context)) {
                 return null;
             }
             Planner.Context plannerContext = context.plannerContext();
-            Plan plan = Merge.ensureOnHandler(
-                plannerContext.planSubRelation(relation.subRelation(), context), plannerContext);
-
-            Limits limits = plannerContext.getLimits(qs);
-            maybeAddFilterProjection(relation, plan);
-            Projection topN = ProjectionBuilder.topNOrEval(
-                relation.subRelation().fields(),
-                qs.orderBy().orElse(null),
+            QueriedRelation subRelation = relation.subRelation();
+            FetchDecider parentFetchDecider = context.fetchDecider();
+            context.setFetchDecider(FetchDecider.NO_FINALIZE);
+            Plan subPlan = plannerContext.planSubRelation(subRelation, context);
+            context.setFetchDecider(parentFetchDecider);
+
+            FetchRewriter.FetchDescription fetchDescription = subPlan.resultDescription().fetchDescription();
+            if (fetchDescription == null) {
+                return finalizeQueriedSelectPlanWithoutFetch(qs, plannerContext, subRelation, subPlan);
+            }
+            return tryToPropagateFetch(qs, subRelation, subPlan, context);
+        }
+    }
+
```

Comment:
pls add a short javadoc telling what this method is doing, and probably also for the others